src/main/java/seedu/loyaltylift/logic/parser/AddOrderCommandParser.java

```diff
@@ -0,0 +1,62 @@
+package seedu.loyaltylift.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.loyaltylift.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.loyaltylift.logic.commands.AddOrderCommand.MESSAGE_USAGE;
+import static seedu.loyaltylift.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.loyaltylift.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.loyaltylift.logic.parser.CliSyntax.PREFIX_QUANTITY;
+
+import java.util.stream.Stream;
+
+import seedu.loyaltylift.commons.core.index.Index;
+import seedu.loyaltylift.logic.commands.AddOrderCommand;
+import seedu.loyaltylift.logic.commands.AddOrderCommand.AddOrderDescriptor;
+import seedu.loyaltylift.logic.parser.exceptions.ParseException;
+import seedu.loyaltylift.model.attribute.Name;
+
+/**
+ * Parses input arguments and creates a new AddOrderCommand object
+ */
+public class AddOrderCommandParser implements Parser<AddOrderCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddOrderCommand
+     * and returns an AddOrderCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddOrderCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME,
+                PREFIX_QUANTITY, PREFIX_ADDRESS);
+        Index customerIndex;
+        Name orderName;
+
+        try {
+            customerIndex = ParserUtil.parseIndex(argMultimap.getPreamble());
+            orderName = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
```

Comment:
Since `name` argument is compulsory, there should be a validation step to check for existence before parsing the argument. 
You can take a look at `AddCustomerCommandParser.java` as a reference